### DISCUSSION TITLE
Quad mesh for Fluent(24R1)

### DISCUSSION THREAD

**Discussion by Schwarz2012** [ansys/pyprimemesh#716] [Q&A] [answered]
Hi, everyone!
I'm new in PyAnsys and maybe it's simple question, but I can't solve it.

I have problem with quad mesh. Simple geometry - cube 50x50x50, no NS.
At first, I tried simple code:
launch prime
read(cube.scdoc)
surface_mesh(max_size = 5)
volume_mesh()
write(cube.msh)
And it's works.
If I change surface_mesh(max_size = 5) to (max_size = 5, generate_quads=True) and use TET in volume mesh - it's  still works, try (quadratic=True) I get error when read .msh file in Fluent:

"Warning: Skipping edges zones section(s).
12378 nodes, binary.
11840 nodes, binary.
Error: Can't read (null) faces into this solver
Clearing partially read grid.
Error at host: Read_Grid_Section: Aborted due to critical error.
Error: Read_Grid_Section: Aborted due to critical error.
Error Object: ()
Error at Node 1: Read_Grid_Section: Aborted due to critical error.
Error at Node 2: Read_Grid_Section: Aborted due to critical error.
Reading Faces: failed while reading section 2013.
Error at Node 0: Read_Grid_Section: Aborted due to critical error.
Error: error: wta[1](string)
Error Object: ()
Error: Error reading "~/cube.msh".
Error Object: #f
Error: api-get-var: the object is not active
Error Object: setup/general/solver/time
Real overflow
Real underflow"

Thanks for any help!

**Comment by waltersma** [comment, marked as answer]
Hi @Schwarz2012 

Welcome to PyAnsys!  

For the final mesh you are creating from your description you will have a quad surface mesh with a layer of pyramids grown from this transitioning to tet elements for the interior.  The elements will all be linear quadratic elements that means you will have mid side nodes on each edge of the elements.  Elements with mid side nodes are not supported in Fluent. 

We can look to catch this as a warning when writing to Fluent formats.

I hope that helps.

Thanks

**Reply by Schwarz2012** [reply]
Hi, @waltersma  
Thank you for answer!

In general, I understand what you mean, but, anyway, what I can do to get full hex mesh in my model, like on screenshot? 
In Ansys Meshing I used "Face Meshing", Solver Preference "Fluent" and it's enough.
![image](https://github.com/ansys/pyprimemesh/assets/119045754/3a893f97-94a7-4d76-8860-4306293e1b4d)

I understand, it's complicated question, but the manual does not provide answers and I would appreciate any help.

**Comment by waltersma** [comment]
Hi @Schwarz2012 ,

As of 0.5.0 there are three different ways you can achieve hex-dominant mesh but each method has its own applicability to a set of cases.

- [Volume sweeper](https://prime.docs.pyansys.com/version/stable/user_guide/stacker.html) if the geometry can be described as 2.5D.  Examples of the usage can be found [here](https://prime.docs.pyansys.com/version/stable/examples/gallery_examples/gallery/05_pcb_stacker.html#sphx-glr-examples-gallery-examples-gallery-05-pcb-stacker-py) and [here](https://prime.docs.pyansys.com/version/stable/examples/gallery_examples/gallery/09_multi_layer_quad_mesh_pcb.html#sphx-glr-examples-gallery-examples-gallery-09-multi-layer-quad-mesh-pcb-py).

- [Thin volume](https://prime.docs.pyansys.com/version/stable/user_guide/automesh.html#thin-volume-mesh-controls) if the volume can be defined as thin and the source faces meshed as quad. An example of the usage can be found [here](https://prime.docs.pyansys.com/version/stable/examples/gallery_examples/gallery/07_saddle_bracket.html#sphx-glr-examples-gallery-examples-gallery-07-saddle-bracket-py).

and 

- [Multizone](https://prime.docs.pyansys.com/version/stable/user_guide/multizone.html) a more general purpose hex-dominant meshing capability, however, this is currently still in Beta which means the API could in theory change before it is fully released.

From the image you shared it is quite hard to propose a way forward but from your description I would expect the use of Multizone to be needed.  Examples will be available once the API is released.

I hope this helps!
